<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1136</width>
    <height>1211</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="7" column="0" colspan="2">
     <widget class="QPushButton" name="startStop">
      <property name="toolTip">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Start / Stop the experiment.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;On start &lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;StopMeNow&lt;/span&gt;=false,&lt;/p&gt;&lt;p&gt;On stop &lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;StopMeNow&lt;/span&gt;=true,&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="text">
       <string>Start/Stop</string>
      </property>
     </widget>
    </item>
    <item row="6" column="0" colspan="2">
     <widget class="QProgressBar" name="progressBar">
      <property name="value">
       <number>0</number>
      </property>
      <property name="format">
       <string>Scan progress: %p% (%v of %m)</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0" colspan="2">
     <layout class="QGridLayout" name="mainLayout">
      <item row="1" column="1">
       <widget class="QFrame" name="frame_2">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="2" column="0">
          <widget class="QWidget" name="placeStepMotor" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_14">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="margin">
             <number>0</number>
            </property>
           </layout>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="nofScansLabel">
           <property name="text">
            <string>Columns</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="stepNof">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;TotalScans&lt;/span&gt;. Shown if &lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;EndCondition&lt;/span&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-style:italic; text-decoration: underline;&quot;&gt;NUMBER&lt;/span&gt;. Otherwise hidden.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QVariantLabel" name="stepCurrent">
           <property name="text">
            <string>&lt;position&gt;</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QTableWidgetWithCopyPaste" name="stepPositionsList">
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="alternatingRowColors">
            <bool>true</bool>
           </property>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
           <column>
            <property name="text">
             <string>Positions</string>
            </property>
           </column>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_34">
           <property name="text">
            <string>Step to move:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_16">
           <item>
            <widget class="QDoubleSpinBox" name="stepSize">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;SerialStep&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="minimum">
              <double>-2147483647.000000000000000</double>
             </property>
             <property name="maximum">
              <double>2147483647.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="irregularOrLabel">
             <property name="text">
              <string>or</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSCheckBox" name="stepIrregular">
             <property name="text">
              <string>Irregular step</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="Line" name="line_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Step motion (horizontal)</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QFrame" name="frame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="3" column="1">
          <widget class="QSpinBox" name="scanRepetitions">
           <property name="minimum">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_47">
           <property name="text">
            <string>Distance</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Repetitions </string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="2" column="1">
          <widget class="QVariantLabel" name="scanCurrent">
           <property name="text">
            <string>&lt;position&gt;</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QWidget" name="placeScanMotor" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_18">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="margin">
             <number>0</number>
            </property>
           </layout>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QDoubleSpinBox" name="scanDistance">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;OutBeam&lt;/span&gt;.  Must not be equal to current position of the &lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;BGmotor&lt;/span&gt; on start if &lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;TotalBGs&lt;/span&gt;&amp;gt;0.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="minimum">
            <double>-2147483647.000000000000000</double>
           </property>
           <property name="maximum">
            <double>2147483647.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="Line" name="line">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QDoubleSpinBox" name="scanSpeed"/>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>Speed</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Scan motion (vertical)</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="2" column="1">
         <widget class="QComboBox" name="shutterSelection"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Shutter</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="browseExpPath">
          <property name="toolTip">
           <string>Browse for the log file</string>
          </property>
          <property name="text">
           <string>Browse</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="expPath"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_14">
          <property name="text">
           <string>Working directory</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_15">
          <property name="text">
           <string>File prefix</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="prefix"/>
        </item>
        <item row="3" column="1">
         <layout class="QGridLayout" name="shutsets">
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineEdit"/>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="lineEdit_3"/>
          </item>
          <item row="1" column="2">
           <widget class="QLineEdit" name="lineEdit_2"/>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="lineEdit_6"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>close</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QLineEdit" name="lineEdit_4"/>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="lineEdit_5"/>
          </item>
          <item row="4" column="2">
           <widget class="QLineEdit" name="lineEdit_8"/>
          </item>
          <item row="3" column="2">
           <widget class="QLineEdit" name="lineEdit_7"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>is open</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>is closed</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>open</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="3" column="2">
         <layout class="QVBoxLayout" name="shuttests">
          <item>
           <widget class="QPushButton" name="pushButton">
            <property name="text">
             <string>close</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_2">
            <property name="text">
             <string>open</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_21">
            <property name="text">
             <string>&lt;status&gt;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_22">
            <property name="text">
             <string>&lt;status&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QFrame" name="scripts_frame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="3" column="1" rowspan="2">
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>scan</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1" rowspan="2">
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>step</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="3">
          <widget class="Line" name="line_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Post</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Post</string>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Post</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_5">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Pre</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="Script" name="preExecScript" native="true"/>
         </item>
         <item row="1" column="2">
          <widget class="Script" name="postExecScript" native="true"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Pre</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0" colspan="3">
          <widget class="Line" name="line_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="Script" name="preScanScript" native="true"/>
         </item>
         <item row="4" column="2">
          <widget class="Script" name="postScanScript" native="true"/>
         </item>
         <item row="6" column="2">
          <widget class="Script" name="preStepScript" native="true"/>
         </item>
         <item row="7" column="2">
          <widget class="Script" name="postStepScript" native="true"/>
         </item>
         <item row="0" column="1" rowspan="2">
          <widget class="QLabel" name="label_11">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>execution</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Pre</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="scripting">
        <property name="text">
         <string>Scripting</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1136</width>
     <height>42</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Script</class>
   <extends>QWidget</extends>
   <header>additional_classes.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QVariantLabel</class>
   <extends>QLabel</extends>
   <header location="global">qtpvwidgets.h</header>
  </customwidget>
  <customwidget>
   <class>QSCheckBox</class>
   <extends>QCheckBox</extends>
   <header>additional_classes.h</header>
  </customwidget>
  <customwidget>
   <class>QTableWidgetWithCopyPaste</class>
   <extends>QTableWidget</extends>
   <header>additional_classes.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>scripting</sender>
   <signal>toggled(bool)</signal>
   <receiver>scripts_frame</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>74</x>
     <y>596</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>677</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
